src/libpgmoneta/wal.c

```diff
@@ -62,6 +62,7 @@ static int wal_xlog_offset(size_t xlogptr, int segsize);
 static int wal_convert_xlogpos(char* xlogpos, uint32_t* high32, uint32_t* low32, int segsize);
 static int wal_find_streaming_start(char* basedir, uint32_t* timeline, uint32_t* high32, uint32_t* low32, int segsize);
 static int wal_shipping_setup(int srv, char** wal_shipping);
+static void update_wal_lsn(char* current_wal_lsn, size_t xlogptr);
```

Comment:
You can resolve `struct configuration` within the function, so no need for a parameter

Reply:
`configuration` can be resolved from `shmem` and we still need `int srv` from `pgmoneta_wal` function. The function signature I can think of is `static void update_wal_lsn(int srv, size_t xlogptr);`. Is this okay?

Reply:
or it should be `static void update_wal_lsn(size_t xlogptr);`?

Reply:
I think it is ok to pass in `int srv`.

Reply:
Yes